### PR TITLE
Update dbt-cloud-features.md

### PR DESCRIPTION
**Pull request by mirnawong1**
remove link to logs per [slack thread](https://dbt-labs.slack.com/archives/C02NCQ9483C/p1678988472058389). will add a new link when [docs deploy IA project](https://www.notion.so/dbtlabs/Reorganize-Deploy-IA-34fc4183cdb7489fbf911eb27f3c5311?pvs=4) is launched.

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/docs/cloud/about-cloud/dbt-cloud-features.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ link="/docs/cloud/develop-in-the-cloud"
 <Card
     title="Manage environments"
     body="Set up and manage separate production and development environments in dbt Cloud to help engineers develop and test code more efficiently, without impacting users or data."
-    link="/docs/collaborate/environments/dbt-cloud-environments"
+    link="/docs/collaborate/environments"
```
Comment (nghi-ly) [inline website/docs/docs/cloud/about-cloud/dbt-cloud-features.md]: i think this is the new URL now. i just checked the docs site. 

```suggestion
    link="/docs/collaborate/environments/environments-in-dbt"
```

**Review by nghi-ly — COMMENTED** [review]
thank you for fixing this! left one sugg.

File: website/docs/docs/cloud/about-cloud/dbt-cloud-features.md (hunk 1 above)
Comment (mirnawong1) [reply website/docs/docs/cloud/about-cloud/dbt-cloud-features.md]: thanks Ly! i guess it changed after i opened this pr on thur

**Review by nghi-ly — APPROVED** [review]
🚢

**Comment by mirnawong1** [comment]
thank you @nghi-ly !!

**Pull request merged**